src/main/java/org/wso2/carbon/mongodb/user/store/mgt/MongoDBUserStoreConstants.java

```diff
@@ -0,0 +1,288 @@
+                "'UM_USER_ID' : '?','UM_ATTR_NAME' : '?','UM_PROFILE_ID' : '?','UM_TENANT_ID' : '?'," +
+                "'projection' : {'$set' : {'UM_ATTR_VALUE' : '?'}}}");
+        setAdvancedProperty(MongoDBCaseInsensitiveConstants.UPDATE_USER_PROPERTY_CASE_INSENSITIVE,
+                MongoDBCaseInsensitiveConstants.UPDATE_USER_PROPERTY_MONGO_CASE_INSENSITIVE);
+        setAdvancedProperty("DeleteUserPropertyMONGO_QUERY", "{'collection' : 'UM_USER_ATTRIBUTE'," +
+                "'UM_USER_ID' : '?','UM_ATTR_NAME' : '?','UM_PROFILE_ID' : '?','UM_TENANT_ID' : '?'}");
+        setAdvancedProperty(MongoDBCaseInsensitiveConstants.DELETE_USER_PROPERTY_CASE_INSENSITIVE,
+                MongoDBCaseInsensitiveConstants.DELETE_USER_PROPERTY_MONGO_CASE_INSENSITIVE);
+        setAdvancedProperty("UserNameUniqueAcrossTenantsMONGO_QUERY", "{'collection' : 'UM_USER_ATTRIBUTE'," +
+                "'UM_USER_ID' : '?','UM_ATTR_NAME' : '?','UM_PROFILE_ID' : '?','UM_TENANT_ID' : '?'}");
+        setAdvancedProperty(MongoDBCaseInsensitiveConstants.USER_NAME_UNIQUE_CASE_INSENSITIVE,
+                MongoDBCaseInsensitiveConstants.USER_NAME_UNIQUE_MONGO_CASE_INSENSITIVE);
+        setAdvancedProperty("IsDomainExistingMONGO_QUERY", "{'collection' : 'UM_DOMAIN','UM_DOMAIN_NAME' : '?'," +
+                "'UM_TENANT_ID' : '?','projection' : {'UM_DOMAIN_ID' : 1}}");
+
+        setAdvancedProperty("AddDomainMONGO_QUERY", "{'collection' : 'UM_DOMAIN','UM_DOMAIN_NAME' : '?'," +
+                "'UM_TENANT_ID' : '?'}");
+
+        setAdvancedProperty("UserSharedRoleMONGO_QUERY", "{'collection' : 'UM_SHARED_USER_ROLE'," +
+                "{'$match' : {'user.UM_USER_NAME' : '?','UM_USER_TENANT_ID' : '?','UM_USER_TENANT_ID' : " +
+                "'$user.UM_TENANT_ID','UM_ROLE_TENANT_ID' : '$role.UM_TENANT_ID'},'$lookup' : '[{'from' : 'UM_USER'," +
+                "'localField' : 'UM_USER_ID','foreignField' : 'UM_ID','as' : 'user'},{'from' : 'UM_ROLE'," +
+                "'localField' : 'UM_ROLE_ID','foreignField' : 'UM_ID','as' : 'roles'}]',{'$unwind' : '$user'}," +
+                "{'$unwind' : '$roles'},'$project' : {'name' : '$_id','UM_ROLE_NAME' : 1,'roles.UM_TENANT_ID' : 1," +
+                "'UM_SHARED_ROLE' : 1}}");
+        setAdvancedProperty("GetUserListOfSharedRoleMONGO_QUERY", "{'collection' : 'UM_SHARED_USER_ROLE',$match : " +
+                "{'UM_ROLE_NAME' : '?','UM_USER_TENANT_ID' : '$user.UM_TENANT_ID','UM_ROLE_TENANT_ID' : " +
+                "'$role.UM_TENANT_ID'},'$lookup' : '[{'from' : 'UM_USER','localField' : 'UM_USER_ID'," +
+                "'foreignField' : 'UM_ID','as' : 'users'},{'from' : 'UM_ROLE','localField' : 'UM_ROLE_ID'," +
+                "'foreignField' : 'UM_ID','$project' : {'name' : '$_id','UM_USER_NAME',1}}]'}");
+    }
+
+    // Private method to set optional properties
+    private static void setProperty(String name, String value, String description) {
```

Comment:
Shall we make use of default java doc comment format for these?

Reply:
Added Java doc comments for all the methods.